scripts/sklearn_training_RFC.py

```diff
@@ -0,0 +1,149 @@
+import re
+import rootpy.io as io
+import rootpy
+from ROOT import TH1D
+import numpy as np
+np.set_printoptions(precision=5)
+import root_numpy as rootnp
+from sklearn.externals import joblib
+log = rootpy.log["/toy_diagnostics"]
+log.setLevel(rootpy.log.INFO)
+
+
+
+#################################
+#				#
+# 	Training		#
+#				#
+#################################
+
+variables = [
+	'trackSip2dSig_0',
+	'trackSip2dSig_1',
+	'trackSip3dSig_0',
+	'trackSip3dSig_1',
+	'trackPtRel_0',
+	'trackPtRel_1',
+	'trackPPar_0',
+	'trackPPar_1',
+	'trackPtRatio_0',
+	'trackJetDist_0',
+	'trackDecayLenVal_0',
+	'vertexEnergyRatio_0',
+	'trackSip2dSigAboveCharm_0',
+	'trackSip3dSigAboveCharm_0',
+	'flightDistance2dSig_0',
+	'flightDistance3dSig_0',
+	'trackSumJetEtRatio',
+	'trackSumJetDeltaR',
+	'massVertexEnergyFraction_0',
+	'leptonRatioRel_0',
+]
+
+
+input_files = [i.strip() for i in open('data/flat_trees/qcd_flat.list')]
+flavors = ['C', 'B', 'DUSG']
+sv_categories = ["NoVertex", "PseudoVertex", "RecoVertex"]
+fname_regex = re.compile('[a-zA-Z_0-9\/]*\/?[a-zA-Z_0-9]+_(?P<category>[a-zA-Z]+)_(?P<flavor>[A-Z]+)\.root')
+
+print 'Merging and converting the samples'
+X = np.ndarray((0,len(variables)),float) # container to hold the combined trees in numpy array structure
+y = np.ones(0) # container to hold the truth signal(1) or background(0) information
+weights = np.ones(0) # container holding the weights for each of the jets
+for fname in input_files:
+   log.info('processing file %s for training' % fname)
+   with io.root_open(fname) as tfile:
+      match = fname_regex.match(fname)
+      if not match:
+         raise ValueError("Could not match the regex to the file %s" % fname)
+      flavor = match.group('flavor')
+      full_category = match.group('category')
+      category = [i for i in sv_categories if i in full_category][0]
+      if flavor == 'B':
+      	log.info('flavour %s is not considered signal or background in this training and is omitted' % flavor) 
+	continue
+      
+      nfiles_per_sample = None
+      skip_n_events = 10 # put this to 1 to include all the events
```

Comment:
It is not really needed indeed, since I am simply merging all the samples. So getting info on the flavour and the category is simply to give some more output that shows which samples are being loaded etc...

Skipping 10 events was for me to make it faster, but I did also notice that the memory usage is going through the roof when using all the events... Also when trying to store the training file once the training is finished, this file can take up a few gigaBytes...